uefi/src/helpers/aligned_buffer.rs

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use alloc::alloc::{alloc, dealloc, Layout, LayoutError};
+use core::error::Error;
+use core::fmt;
+
+/// Helper class to maintain the lifetime of a memory region allocated with a non-standard alignment.
+/// Facilitates RAII to properly deallocate when lifetime of the object ends.
+///
+/// Note: This uses the global Rust allocator under the hood.
+#[allow(clippy::len_without_is_empty)]
+#[derive(Debug)]
+pub struct AlignedBuffer {
+    ptr: *mut u8,
+    layout: Layout,
+}
```

Comment:
nit: add a blank line after a `}` that closes a struct or function (so, also add blank line separating each method in the `impl` below)